src/Makefile

```diff
@@ -1157,22 +1157,30 @@ CPYTHON := cpython
 PYTHON_SOURCE := no
 
 ifneq (${TARGET}, winagent)
-cpu_arch := ${uname_M}
-ifneq (,$(filter ${cpu_arch},x86_64 amd64))
-PRECOMPILED_ARCH := /amd64
-else
-ifneq (,$(filter ${cpu_arch},i386 i686))
-PRECOMPILED_ARCH := /i386
-else
-ifneq (,$(filter ${cpu_arch},aarch64 arm64))
-PRECOMPILED_ARCH := /aarch64
+ifneq (,$(filter ${uname_S}, Linux Darwin))
+	cpu_arch := ${uname_M}
+	ifneq (,$(filter ${cpu_arch},x86_64 amd64))
+		PRECOMPILED_ARCH := /amd64
+	else
+	ifneq (,$(filter ${cpu_arch},i386 i686))
+		PRECOMPILED_ARCH := /i386
+	else
+	ifneq (,$(filter ${cpu_arch},aarch64 arm64))
+		PRECOMPILED_ARCH := /aarch64
+	else
+	ifneq (,$(filter ${cpu_arch},armv8l armv7l arm32 armhf))
+		PRECOMPILED_ARCH := /arm32
+	else
+		PRECOMPILED_ARCH := /${uname_P}
+	endif
+	endif
+	endif
+	endif
 else
-ifneq (,$(filter ${cpu_arch},armv8l armv7l arm32 armhf))
-PRECOMPILED_ARCH := /arm32
+ifneq (,$(filter ${uname_S},HP-UX SunOS AIX))
```

Comment:
Just a comment, I was thinking about this structure in fact

```bash
ifneq (,$(filter ${uname_S},Linux Darwin HP-UX))
	cpu_arch := ${uname_M}
	ifneq (,$(filter ${cpu_arch},x86_64 amd64))
		PRECOMPILED_ARCH := /amd64
	else
	ifneq (,$(filter ${cpu_arch},i386 i686))
		PRECOMPILED_ARCH := /i386
	else
	ifneq (,$(filter ${cpu_arch},aarch64 arm64))
		PRECOMPILED_ARCH := /aarch64
	else
	ifneq (,$(filter ${cpu_arch},armv8l armv7l arm32 armhf))
		PRECOMPILED_ARCH := /arm32
        else
        ifeq (${cpu_arch},not)                        #NOTE: it should be 'ia64', instead of 'not'
		PRECOMPILED_ARCH := /not
	else
		PRECOMPILED_ARCH := /${uname_P}
	endif
	endif
	endif
	endif
	endif
else
ifneq (,$(filter ${uname_S},SunOS AIX))
	cpu_arch := ${uname_P}
        ifeq (${cpu_arch},powerpc)  
		PRECOMPILED_ARCH := /powerpc
	else
	ifneq (,$(filter ${cpu_arch},sparc sun4u))
		PRECOMPILED_ARCH := /sparc
	else
	ifneq (,$(filter ${cpu_arch},i386 i86pc))
		PRECOMPILED_ARCH := /i386
	else
		PRECOMPILED_ARCH := /${uname_M}
	endif
	endif
	endif
	endif
else
    PRECOMPILED_ARCH := /${uname_M}
endif
endif
```

Reply:
That's good. I used your suggestion.

I am not so sure we should use the fallbacks if the resulted value does not match one of the supported 'os/arch' paths. It is not critical because the dependencies will end up being built from sources, but it would nice to let the user know about that. Plus avoid a lot of downloading error messages and save some time. Anyway, I think that is out of the scope of this issue.

Reply:
> but it would nice to let the user know about that. Plus avoid a lot of downloading error messages and save some time. Anyway, I think that is out of the scope of this issue.

I totally agree. But that is a bigger rework, it's out of scope as you say.
The only simple think I cam come out it's the use of many **echo** messages:
- "Architecture found: X" or "Architecture not found, trying: X"
- "Trying to download pre-compiled sources for 'X' with arch 'X'"
- "pre-compiled dependencies failed, trying to compile from sources now"
- etc.

But it still may be implemented in another issue.